### DISCUSSION TITLE
Lets collect all our ideas for new features here

### DISCUSSION THREAD

**Discussion by d3vilh** [d3vilh/openvpn-ui#51] [Ideas]
Hey community,
If you have any new Idea, please share it here. So, we can continue improve UI and include new features.

**Comment by wolf14sk** [comment]
Hello. It would be nice if the option to add and edit "client-specific" settings, which are located in the "staticclients" folder, could be added through the UI.
Thank you

**Reply by jannoke** [reply]
I would except some readymade textboxes for most used things like:

- route networks behind this host (iroute)
- allow this network accesss to these networks (route)
- i guess static ip should be there also, but I did not see it in there (ifconfig-push #.#.#.# #.#.#:#)

**Reply by d3vilh** [reply]
> * i guess static ip should be there also, but I did not see it in there (ifconfig-push #.#.#.# #.#.#:#)

OpenVPN server have standard mechanism to define static client IPs via client configuration files located in dedicated subdirectory (option `client-config-dir /etc/openvpn/staticclients` in `server.conf`)

Here is example from my test env:
```bash
getman@Ukraine:~/build/openvpn-server $ cd staticclients/ && ls -lrt
total 4
-rw-r--r-- 1 root root 39 May 11 20:28 test.static.ip
getman@Ukraine:~/build/openvpn-server/staticclients $ cat test.static.ip
ifconfig-push 10.0.70.50 255.255.255.0
getman@Ukraine:~/build/openvpn-server/staticclients $
```

You could change IPs there.

Like this idea too.
I will think on how can we read this dynamically on `Certificates` page for users which is already exists and update this IPs right way (not via `index` file).

**Comment by jannoke** [comment]
About static ip setting:
- There should be defined pools for static and non-static ranges which will be checked in ui. Currently you can put whatever static ip in client field and it is not checked if it's in range. 
- Currently you can give multiple clients same static ip and there is no verification, which could have unpredicatble results
- Seems like static ip can't be changed later?

**Comment by d3vilh** [comment]
> * Currently you can give multiple clients same static ip and there is no verification, which could have unpredicatble results
I think if we decide to keep "guest" and "allowed" users subnets permanent, then I can easy add the dropdown or other type of menu to choose the not used permanent IP.

I like this request will think how can we do it. 

💡Maybe we can add those subnets to OpenVPN-UI vars as internal BeeGo app vars. In that case these subnets can be dynamic.
